### PR TITLE
feat: progress: add progress component to octuple

### PR DESCRIPTION
**Pull request by dkilgore-eightfold**
## SUMMARY:
This change adds the Progress component to Octuple.

-Also consolidates reset component and clear-fix mixins, fixes tree import type so the filter tree expand/collapse actually works.

https://user-images.githubusercontent.com/99700808/195478194-65e5b584-223b-4180-91fd-9bb6556fea93.mp4

Octuple DS Figma spec: https://www.figma.com/file/SlKRC7oKF7XZyHMv2op4ch/Octuple-DS-(Theme-2)?node-id=7994%3A24301

## JIRA TASK (Eightfold Employees Only):
ENG-2822

## CHANGE TYPE:

-   [ ] Bugfix Pull Request
-   [X] Feature Pull Request

## TEST COVERAGE:

-   [ ] Tests for this change already exist
-   [X] I have added unittests for this change

## TEST PLAN:
Pull the pr feature branch, then do `yarn` and `yarn storybook`. Verify the Progress stories behave as expected.

### CODE REVIEW

File: src/components/Progress/Steps.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        }
+    };
+
+    useLayoutEffect(() => {
+        updateLayout();
+    }, [showLabels, showValueLabel]);
+
+    return (
+        <ResizeObserver onResize={updateLayout}>
+            <div ref={flexContainerRef} className={styles.progressStepsOuter}>
```
Comment (ychhabra-eightfold) [inline src/components/Progress/Steps.tsx:139]: are these accessable ? Same question for all the other components

**Review by ychhabra-eightfold — CHANGES_REQUESTED** [review]

File: src/components/Progress/Circle.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    const { strokeColor: successColor } = success;
+    return [successColor || 'var(--success-color)', strokeColor || null!];
+}
+
+const Circle: FC<CircleProps> = (props) => {
+    const {
+        children,
+        gapDegree,
+        gapPosition,
+        strokeLinecap = 'butt',
```
[the remark this answers is not in the document]
Comment (dkilgore-eightfold) [reply src/components/Progress/Circle.tsx:30]: ya, funny name, but it's correct.

File: src/components/Progress/Internal/hooks/useId.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { canUseDom } from '../../../../shared/utilities';
+
+let uuid: number = 0;
+
+/** Is client side and not jsdom */
+export const isBrowserClient = process.env.NODE_ENV !== 'test' && canUseDom();
+
+/** Get unique id for accessibility usage */
```
[the remark this answers is not in the document]
Comment (dkilgore-eightfold) [reply src/components/Progress/Internal/hooks/useId.ts:9]: Yes, this is built on the global uid

File: src/components/Progress/Steps.tsx (hunk 1 above)
Comment (dkilgore-eightfold) [reply src/components/Progress/Steps.tsx:139]: The text content is accessible. The visual representations would only need tab indices and aria label props. I'll add these.

File: src/components/Progress/Circle.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        strokeWidth = VIEW_BOX_SIZE / 1.6,
+        success,
+        trailColor = null as any,
```
[the remark this answers is not in the document]
Comment (dkilgore-eightfold) [reply src/components/Progress/Circle.tsx:33]: TypeScript nullable string is not supported. I could do a union in the types file to handle this.

File: src/components/Progress/Internal/OcLine.tsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    strokeLinecap,
+    strokeWidth,
+    style,
+    trailColor,
+    trailWidth,
+    transition,
+    ...rest
+}) => {
+    // eslint-disable-next-line no-param-reassign
+    delete rest.gapPosition;
```
[the remark this answers is not in the document]
Comment (dkilgore-eightfold) [reply src/components/Progress/Internal/OcLine.tsx:21]: Due to shared default props in common, may remove when separated.

File: src/components/Progress/Steps.tsx (hunk 1 above)
Comment (dkilgore-eightfold) [reply src/components/Progress/Steps.tsx:139]: To answer your question. The Octuple component library is accessible. However, there will be bugs (as always) that are device specific - iOS, Android, Windows, MacOS, Ubuntu, etc... all use different accessible technologies and the test matrix is massive due to unique versions, browsers and more. Much too great to cover 100% in 4 months time. I will say this however, the last product I worked on was a gleaming example of A11y support for hundreds of millions of users (MAU/DAU). I'm striving to apply those best practices here, though it is a living effort and will never be 'complete'.

Comment (dkilgore-eightfold) [reply src/components/Progress/Steps.tsx:139]: Now if we could migrate vscode app to use Layout, Grid, and Stack to support A11y regions that would be dope.

File: src/components/Progress/Internal/OcLine.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (dkilgore-eightfold) [reply src/components/Progress/Internal/OcLine.tsx:21]: Good trick though, `delete`, something to put in our pockets for later :)

File: src/components/Progress/Steps.tsx (hunk 1 above)
Comment (dkilgore-eightfold) [reply src/components/Progress/Steps.tsx:139]: No A11y violations found using the addon.

File: src/styles/abstracts/_mixins.scss [hunk 6]
```diff
@@ -19,6 +19,31 @@
...
+@mixin reset-component() {
+    box-sizing: border-box;
+    margin: 0;
+    padding: 0;
+    color: var(--text-primary-color);
+    font-size: $text-font-size-2;
+    font-variant: tabular-nums;
+    line-height: 1.5715;
+    list-style: none;
+    font-feature-settings: 'tnum';
```
[the remark this answers is not in the document]
Comment (dkilgore-eightfold) [reply src/styles/abstracts/_mixins.scss:31]: tnum enables tabular monospace figures. https://it4np.com/2016/12/use-tabular-figures-aka-monospaced-numbers-in-tables-rather-than-proportional-digits.html

File: src/styles/abstracts/_mixins.scss [hunk 7]
```diff
@@ -19,6 +19,31 @@
...
+}
+
+@mixin clearfix() {
```
[the remark this answers is not in the document]
Comment (dkilgore-eightfold) [reply src/styles/abstracts/_mixins.scss:34]: Used by a couple of the more convoluted fixed layouts. Will investigate ways to remove. It's unrelated to this change, but figured I'd move it to mixins.

File: src/styles/abstracts/_mixins.scss [hunk 8]
```diff
@@ -19,6 +19,31 @@
     opacity: 0;
 }
 
+@mixin reset-component() {
```
[the remark this answers is not in the document]
Comment (dkilgore-eightfold) [reply src/styles/abstracts/_mixins.scss:22]: tnum only enables a font feature that may be used. Overall this mixin does what it implies. tabular-nums is a good css trick to ensure alignment of numbers. Here's a nice tweet explaining --> https://twitter.com/javan/status/1486059026064584711

**Review by ychhabra-eightfold — APPROVED** [review]

**Pull request merged**